core/node/eth_watch/src/event_processors/interop_chain_batch_root.rs

```diff
@@ -0,0 +1,155 @@
+        let events_count = events.len();
+        let mut transaction = storage
+            .start_transaction()
+            .await
+            .map_err(DalError::generalize)?;
+
+        let sl_chain_id = self.inner.sl_l2_client.chain_id().await?;
+        for event in events {
+            let sl_block_number = L2BlockNumber(
+                event
+                    .block_number
+                    .expect("Missing block number for finalized event")
+                    .as_u32(),
+            );
+            let batch_number = L1BatchNumber(
+                event
+                    .l1_batch_number
+                    .expect("Missing L1 batch number for finalized event")
+                    .as_u32(),
+            );
+
+            let batch_proof = {
+                let leaf_position = self.inner.merkle_tree.length() - 1;
+                let batch_proof = self
+                    .inner
+                    .merkle_tree
+                    .merkle_root_and_path_by_absolute_index(leaf_position)
+                    .1;
+                let batch_proof_len = batch_proof.len() as u32;
+                let mut proof = vec![H256::from_low_u64_be(leaf_position as u64)];
+                proof.extend(batch_proof);
+
+                BatchAndChainMerklePath {
+                    batch_proof_len,
+                    proof,
+                }
+            };
+
+            let chain_agg_proof_until_msg_root = self
+                .inner
+                .sl_l2_client
+                .get_chain_log_proof_until_msg_root(sl_block_number, self.inner.l2_chain_id)
+                .await?
+                .context("Missing chain log proof until msg root for finalized batch")?;
+            let chain_proof_vector_until_msg_root = Self::chain_proof_vector(
+                sl_block_number,
+                chain_agg_proof_until_msg_root,
+                sl_chain_id,
+            );
+
+            let mut batch_chain_proof_until_msg_root = batch_proof;
+            batch_chain_proof_until_msg_root
+                .proof
+                .extend(chain_proof_vector_until_msg_root);
+            transaction
+                .blocks_dal()
+                .set_batch_chain_merkle_path_until_msg_root(
+                    batch_number,
+                    batch_chain_proof_until_msg_root,
+                )
+                .await
+                .map_err(DalError::generalize)?;
+        }
+
+        transaction.commit().await.map_err(DalError::generalize)?;
+
+        Ok(events_count)
+    }
+
+    fn topic1(&self) -> Option<H256> {
+        Some(self.inner.appended_chain_batch_root_signature)
+    }
+
+    fn topic2(&self) -> Option<H256> {
+        Some(H256::from_low_u64_be(self.inner.l2_chain_id.as_u64()))
+    }
+
+    fn event_source(&self) -> EventsSource {
+        EventsSource::SL
+    }
+
+    fn event_type(&self) -> EventType {
+        EventType::ChainBatchRoot
+    }
+
+    fn only_finalized_block(&self) -> bool {
```

Comment:
we want this to be not finalized, right? I'm unsure atm ( did not think it fully through)

Reply:
why do we even need this?

Reply:
We need to set save two distinct merkle paths for interop vs withdrawals.